From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Rishwanth Yeddula <rish@cpanel.net>
Date: Mon, 2 Oct 2017 13:59:13 -0500
Subject: [PATCH 4/8] Suppress logging of empty messages

---
 src/cxx_supportlib/LoggingKit/Implementation.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/cxx_supportlib/LoggingKit/Implementation.cpp b/src/cxx_supportlib/LoggingKit/Implementation.cpp
index ba919b4..cd04254 100644
--- a/src/cxx_supportlib/LoggingKit/Implementation.cpp
+++ b/src/cxx_supportlib/LoggingKit/Implementation.cpp
@@ -438,6 +438,9 @@ logAppOutput(const HashedStaticString &groupName, pid_t pid, const StaticString
 	int targetFd;
 	bool saveLog = false;
 
+	if (size == 0) {
+		return;
+	}
 	if (OXT_LIKELY(context != NULL)) {
 		const ConfigRealization *configRealization = context->getConfigRealization();
 		if (configRealization->level < configRealization->appOutputLogLevel) {
